blockchain/filedao/filedao.go

```diff
@@ -281,9 +357,14 @@ func CreateFileDAO(legacy bool, gen genesis.Genesis, cfg config.DB) (FileDAO, er
 	}
 
 	// populate v2 files into v2 manager
-	for _, name := range v2Files {
-		cfg.DbPath = name
-		fds = append(fds, openFileDAOv2(cfg))
+	if len(v2Files) > 0 {
+		for _, name := range v2Files {
+			cfg.DbPath = name
+			fds = append(fds, openFileDAOv2(cfg))
+		}
+
+		// v2 file's top index overrides v1's top
+		fd.topIndex = v2Top
```

Comment:
for example, we also have 2 v2 files chain-00000003.db, chain-00000004.db
then fd.topIndex = 4